docs/design/incrementaldownload.md

```diff
@@ -0,0 +1,66 @@
+# Incremental Download
+
+## Overview
+The feature is to allow Incremental Download of the Artifact in release so that all of the artifact items are not downloaded from the source provider and can be retrieved from an artifact cache maintained on the agent.
+
+If services storing builds provide hash of the artifact file content, it can be leveraged to perform incremental download of artifacts. Since most of the popular services (Jenkins/TeamCity) do not provide file content hashes, solution is to generate the hashes before publishing the artifact and use this information during artifact download.
+
+## Task Changes
+A new task will be created named `Generate Artifact Metadata` that will generate `artifact-metadata.csv` file containing list of file paths and their file content hashes mapped to one another, created in the same directory as that of the artifacts.
```

Comment:
mention that this might collide in the rare case where artifact already has artifact-metadata.csv file, but we can certainly handle it in future or work around.